packages/core/contracts/zodiac/OptimisticOracleModule.sol

```diff
@@ -0,0 +1,271 @@
+            if (_transactions[i].data.length > 0) {
+                require(_isContract(_transactions[i].to), "EOA can't accept tx with data");
+            }
+            proposal.transactions.push(_transactions[i]);
+        }
+
+        // Get the bond from the proposer and approve the bond and final fee to be used by the oracle.
+        uint256 totalBond = finalFee + bond;
+        collateral.safeTransferFrom(msg.sender, address(this), totalBond);
+        collateral.safeIncreaseAllowance(address(skinnyOptimisticOracle), totalBond);
+
+        // Propose a set of transactions to the OO. If not disputed, they can be executed with executeProposal().
+        // docs: https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/oracle/interfaces/SkinnyOptimisticOracleInterface.sol
+        skinnyOptimisticOracle.requestAndProposePriceFor(
+            identifier,
+            uint32(time),
+            proposal.ancillaryData,
+            collateral,
+            0,
+            bond,
+            uint256(liveness),
+            proposer,
+            // Canonical value representing "True"; i.e. the transactions are valid.
+            int256(1e18)
+        );
+
+        emit TransactionsProposed(id, proposer, time);
+    }
+
+    function executeProposal(uint256 _proposalId, uint256 _transactionIndex) public payable nonReentrant {
+        // Execute transactions in an approved proposal using exec() function.
+        Proposal storage proposal = proposals[_proposalId];
+
+        // This will revert if the price has not settled.
+        int256 price = _getOracle().getPrice(identifier, proposal.requestTime, proposal.ancillaryData);
+
+        Transaction memory transaction = proposal.transactions[_transactionIndex];
+
+        require(
+            _transactionIndex == 0 || proposal.transactions[_transactionIndex - 1].to == address(0),
+            "Previous tx not yet executed"
+        );
+        require(transaction.to != address(0), "Tx already executed");
+        require(price == 1e18, "Proposal was rejected");
+        require(msg.value == transaction.value, "Must send exact amount of ETH");
+
+        // Delete the transaction before execution to avoid any potential re-entrancy issues.
+        delete proposal.transactions[_transactionIndex];
+
+        require(
+            exec(transaction.to, transaction.value, transaction.data, transaction.operation),
+            "Failed to execute the transaction"
+        );
+        emit TransactionExecuted(_proposalId, _transactionIndex);
+    }
+
+    function deleteProposal(uint256 _proposalId) public onlyOwner {
+        // Delete a proposal that governance decided not to execute.
+        delete proposals[_proposalId];
+        emit ProposalDeleted(_proposalId);
+    }
+
+    function deleteRejectedProposal(uint256 _proposalId) public {
+        // Execute transactions in an approved proposal using exec() function.
```

Comment:
This comment is misleading here as deleting the proposal should not execute its transactions

Reply:
Whoops, will delete.